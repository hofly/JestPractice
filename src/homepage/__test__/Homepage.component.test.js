import React from 'react';
import { shallow } from 'enzyme'
import TestRenderer from "react-test-renderer";

import Homepage from './../Homepage.component';

describe('Homepage Test', () => {
    it('should render snapshot test for homepage', () => {
        // const wrapper = TestRenderer.create(<Homepage />).toJSON();
        // expect(wrapper).toMatchSnapshot()

        const wrapper = shallow(<Homepage />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should show state as message expect when mount component', () => {
        const wrapper = shallow(<Homepage />)
        console.log(wrapper.state())
        expect(wrapper.state().title).toBe('Welcome to user board')
    })

    it('should navigate to anonymous users', () => {
        let props = {navigation: {navigate: jest.fn()}}

        const wrapper = shallow(<Homepage {...props}/>)

        // const instance = wrapper.instance()

        // const spy=jest.spyOn(instance, 'navigateToAnonymousUser')
        // instance.forceUpdate(); //call forceUpdate to attach the spy function to the instance

        wrapper.find('TouchableOpacity').at(0).simulate('press')

        // expect(spy).toBeCalled()

        expect(props.navigation.navigate).toBeCalledWith('user', { userType: 'anonymous' })
    })

    it('should turn trial version when navigating to anonymous user', () => {
        let props = {navigation: {navigate: jest.fn()}}
        const wrapper = shallow(<Homepage {...props}/>)
        wrapper.find('TouchableOpacity').at(0).simulate('press')
        console.log(wrapper.state())
        expect(wrapper.state().isTrial).toBe(true)
    })

    it('should navigate to professional users', () => {
        let props = {navigation: {navigate: jest.fn()}}
        const wrapper = shallow(<Homepage {...props}/>)
        wrapper.find('TouchableOpacity').at(1).simulate('press')
        expect(props.navigation.navigate).toBeCalledWith('user', { userType: 'professional' })
    })

    it('should navigate to premium users', () => {
        let props = {navigation: {navigate: jest.fn()}}
        const wrapper = shallow(<Homepage {...props}/>)
        wrapper.find('TouchableOpacity').at(2).simulate('press')
        expect(props.navigation.navigate).toBeCalledWith('user', { userType: 'premium' })
    })
})