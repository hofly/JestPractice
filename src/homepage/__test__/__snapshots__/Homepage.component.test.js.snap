// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Test should render snapshot test for homepage 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Welcome to user board
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 20,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9CCC65",
            "borderRadius": 15,
            "marginBottom": 10,
            "paddingVertical": 15,
            "width": 200,
          }
        }
      >
        <Text>
          Anonymous Users
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9CCC65",
            "borderRadius": 15,
            "marginBottom": 10,
            "paddingVertical": 15,
            "width": 200,
          }
        }
      >
        <Text>
          Professional Users
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#9CCC65",
            "borderRadius": 15,
            "marginBottom": 10,
            "paddingVertical": 15,
            "width": 200,
          }
        }
      >
        <Text>
          Premium Users
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
